sw/device/lib/crypto/impl/ecc.c

```diff
@@ -170,7 +170,7 @@ static status_t key_config_check(const ecc_curve_t *elliptic_curve,
   }
 
   // Should never get here.
-  HARDENED_UNREACHABLE();
+  HARDENED_TRAP();
   return OTCRYPTO_FATAL_ERR;
```

Comment:
Is this now going to generate the code for the return statement whereas before it wouldn't? Do we want that for the extra-extra hardening reassurance? (I don't know if we have documented such fine-grained security/hardening guidelines, to clarify the policy on these kinds of tradeoffs).

Reply:
I'll let @jadephilipoom answer this one, typically we just omit `return`s after `HARDENED_TRAP()` in silicon_creator code since you can't test it on or off target.

Reply:
> typically we just omit `return`s after `HARDENED_TRAP()` in silicon_creator code since you can't test it on or off target.

What do you mean? If's it's just a trap (i.e. not marked as unreachable) isn't that going to fail to compile?

Reply:
> What do you mean? If's it's just a trap (i.e. not marked as unreachable) isn't that going to fail to compile?

Of course, I was referring to the case before this PR: we did not have any `return` statements after `HARDENED_UNREACHABLE()` and there were no errors because of the `__builtin_unreachable()` at the end of that macro. After the recent changes, this PR replaces all of those with `HARDENED_TRAP()` and adds `OT_UNREACHABLE()`s wherever they are needed.